format/Layout.md

```diff
@@ -135,21 +135,18 @@ Unless otherwise noted, padded bytes do not need to have a specific value.
 
 ## Array lengths
 
-Any array has a known and fixed length, stored as a 32-bit signed integer, so a
-maximum of 2<sup>31</sup> - 1 elements. We choose a signed int32 for a couple reasons:
-
-* Enhance compatibility with Java and client languages which may have varying
-  quality of support for unsigned integers.
-* To encourage developers to compose smaller arrays (each of which contains
-  contiguous memory in its leaf nodes) to create larger array structures
-  possibly exceeding 2<sup>31</sup> - 1 elements, as opposed to allocating very large
-  contiguous memory blocks.
+Array lengths are represented in the Arrow metadata as a 64-bit signed
+integer. An implementation of Arrow is considered valid even if it only
+supports lengths up to the maximum 32-bit signed integer, though. If using
+Arrow in a multi-language environment, we recommend limiting lengths to
+2<sup>31</sup> - 1 elements or less.
```

Comment:
Perhaps mention that the limitation can be overcome by having multiple array chunks?

Reply:
done! this will help with follow up questions